### ISSUE TITLE
Non-breaking space is removed when another non-breaking space is added

### ISSUE THREAD

**Issue by tomsq** [tinymce/tinymce#9228]
## 📝 Provide detailed reproduction steps (if any)
Adding additional breaking space removes previously added breaking space, you can see that clearly in the following repro: 
https://codesandbox.io/p/sandbox/tinymce-removes-nbsp-j9v3dh?file=%2Findex.html 

Step 1: Add non-breaking space to the end of the 2nd line.
Step 2: Add another non-breaking space somewhere else

First non-breaking space is removed.

### ✔️ Expected result
Both breaking spaces are present.

### ❌ Actual result
Only the 2nd non-breaking space is present.

### ❓ Possible solution
The workaround is to set `nonbreaking_wrap` to `true` but I need to remove the spans later to prevent a lot html tags because of spam tests.

## 📃 Other details
- Browser: Chrome 119
- OS: macOS
---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by lostkeys** [comment]
@tomsq Sorry, I'm not able to replicate your issue. Your description mentions a second line, but the replication demo only has one line of text. Regardless of what I try, all `&nbsp;` are retained and nothing is removed.

**Comment by tomsq** [comment]
@lostkeys Apologize for not providing clear instructions, here is a GIF of what I mean. \
![breaking-space-removed](https://github.com/tinymce/tinymce/assets/25229106/b1cebd50-deb8-46e9-beb7-d606c684519a)

**Comment by lostkeys** [comment]
I can confirm this being an issue. Here is a reduced replication case: https://fiddle.tiny.cloud/OgrDglWjwK/1. This have been an issue for years, dating back to TinyMCE 4.9

**Comment by jdus** [comment]
Absolutely. This issue has been existing for a long time, as also reported by me: https://github.com/tinymce/tinymce/issues/7114
Wrapping nbsp in spans helps, but is not always possible nor beautiful...

**Comment by didierbrouwers** [comment]
This bug still exists in TinyMCE 6. Are there any updates?
